main.tf

```diff
@@ -103,3 +106,35 @@ module "cbr_rule" {
     }]
   }]
 }
+
+##############################################################################
+# Event Notification
+##############################################################################
+
+# Create IAM Authorization Policies to allow SM to access event notification
+resource "ibm_iam_authorization_policy" "en_policy" {
+  count                       = var.enable_event_notification == false || var.skip_en_iam_authorization_policy ? 0 : 1
+  source_service_name         = "secrets-manager"
+  source_resource_group_id    = var.resource_group_id
+  target_service_name         = "event-notifications"
+  target_resource_instance_id = regex(".*:(.*)::", var.existing_en_instance_crn)[0]
+  roles                       = ["Event Source Manager"]
+  description                 = "Allow all Secrets Manager instances in the resource group ${var.resource_group_id} to access the event-notification instance GUID ${regex(".*:(.*)::", var.existing_en_instance_crn)[0]}"
+}
+
+# workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478
+resource "time_sleep" "wait_for_en_authorization_policy" {
+  depends_on = [ibm_iam_authorization_policy.en_policy]
+
+  create_duration = "30s"
+}
+
+resource "ibm_sm_en_registration" "sm_en_registration" {
+  count                                  = var.enable_event_notification ? 1 : 0
+  depends_on                             = [time_sleep.wait_for_en_authorization_policy]
+  instance_id                            = local.secrets_manager_guid
+  region                                 = var.region
+  event_notifications_instance_crn       = var.existing_en_instance_crn
+  event_notifications_source_description = "Secret Manager"
+  event_notifications_source_name        = var.secrets_manager_name
```

Comment:
we probably want to expose [endpoint_type](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/sm_en_registration#endpoint_type) here so consumer can decide if they want to use the public or private endpoint